part2/stages/Commit-preserved-data

```diff
@@ -26,8 +26,16 @@ for FILE in ${ALL_DOM0_CONFIGS} ; do
     do_cmd mv "${FILE}.new" "${FILE}" || exit ${Abort}
 done
 
-# Fix SELinux labeling in /boot/system/config
-do_cmd chroot ${DOM0_MOUNT} sh -c "/sbin/setfiles /etc/selinux/xc_policy/contexts/files/file_contexts /boot/system/config" >&2
+ARCH=$(uname -m)
+if [ "$ARCH" = "i686" ]; then
+    # Fix SELinux labeling in /boot/system/config
+    # If we're upgrading from 32-bit we need to use our compat image
+    # It should already be mounted.
+    do_cmd chroot ${MOUNT_ROOT}/upgrade sh -c "/sbin/setfiles /etc/selinux/xc_policy/contexts/files/file_contexts /boot/system/config" >&2
```

Comment:
Switch "${MOUNT_ROOT}/upgrade" to ${UPGRADE_MOUNT} from file-paths?  Should already be sourced in through `. ${SCRIPT_DIR}/functions`

Reply:
Good call, missed that.